java/show-labels-on-layer/README.md

```diff
@@ -1,26 +1,43 @@
-# Show Labels on Layer
-Demonstrates how to show labels on a feature layer.
+# Show labels on layer
 
-![Show Labels on Layer App](show-labels-on-layer.png)
+Display custom labels on a feature layer.
+
+![Image of show labels on layers](show-labels-on-layer.png)
+
+## Use case
+
+Labeling features is useful to visually display a key piece of information or attribute of a feature on a map. For example, you may want to label rivers or streets with their names. 
 
 ## How to use the sample
-Simply load the sample. Notice the way labels change with different zoom levels.
+
+Pan and zoom around the United States. Labels for congressional districts will be shown in red for Republican districts and blue for Democrat districts. Notice how labels pop into view as you zoom in.
 
 ## How it works
 
-1. Create a `FeatureLayer` with a `ServiceFeatureTable` using an online feature service.
-1. Create a `TextSymbol` to use for displaying the label text.
-    - Create a JSON string for the label definition.
-    - Set the "LabelExpressionInfo.expression" key to express what the text the label should display. You can use fields of the feature by using `$feature.field_name` in the expression.
-    - To use the text symbol, set the "symbol" key to the symbol's JSON representation using `textSymbol.toJson()`.
-1. Create a label definition from the JSON using `LabelDefinition.fromJson(json)`.
-1. Add the definition to the feature layer with `featureLayer.getLabelDefinitions().add(labelDefinition)`.
-1. Lastly, enable labels on the layer using `featureLayer.setLabelsEnabled()`.
+1. Create a `ServiceFeatureTable` using a feature service URL.
+2. Create a `FeatureLayer` from the service feature table.
+3. Create a `TextSymbol` to use for displaying the label text.
+4. Create a JSON string for the label definition.
+    * Set the "LabelExpressionInfo.expression" key to express what the text the label should display. You can use fields of the feature by using `$feature.field_name` in the expression.
+    * To use the text symbol, set the "symbol" key to the symbol's JSON representation using `textSymbol.toJson()`.
+5. Create a label definition from the JSON using `LabelDefinition.fromJson(json)`.
```

Comment:
perhaps worth mentioning (for both java and kotlin): the sample uses `LabelDefinition.fromJson(json.toString())`

Reply:
Good catch, I'll update the call to `LabelDefinition.fromJson(jsonString)`